storage/range_test.go

```diff
@@ -260,12 +254,28 @@ func TestRangeContains(t *testing.T) {
 	}
 }
 
-func TestRangeCanService(t *testing.T) {
+func setLeaderLease(t *testing.T, r *Range, l *proto.Lease) {
+	req := &proto.InternalLeaderLeaseRequest{Lease: *l}
+	reply := &proto.InternalLeaderLeaseResponse{}
+	if err := r.AddCmd(req, reply, true); err != nil {
+		t.Errorf("failed to set lease: %s", err)
+	}
+}
+
+// TestRangeReadConsistency verifies behavior of the range under
+// different read consistencies. Note that this unittest plays
+// fast and loose with granting leader leases.
+func TestRangeReadConsistency(t *testing.T) {
 	defer leaktest.AfterTest(t)
 	tc := testContext{}
 	tc.Start(t)
 	defer tc.Stop()
 
+	pArgs, pReply := putArgs([]byte("a"), []byte("value"), 1, tc.store.StoreID())
```

Comment:
Too many successive changes to these tests over the course of this PR. It's unnecessary and has been removed. The leader lease is acquired on the consistent Get which comes directly after the put.